### PR TITLE
[Docs] Out of place esimd radix sort documentation update

### PR DESCRIPTION
**Pull request by danhoeflinger**
This PR updates the esimd radix sort documentation with the out of place overloads which were added in #1439.

It also cleans up some minor typos and fixes the namespace in the examples to match their new home after the move.

The memory requirements and performance considerations for out of place overloads should be equivalent to the in place overloads.  Out of place overloads run the same kernel code internally, it just provides different input and output sequences to the kernels to allow users to 
(1) preserve input sequence order
(2) fold in a copy to an output sequence

### CODE REVIEW

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dcbenito) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:255]: Please change line five as well, to remove the monospace formatting.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dcbenito) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:256]: The header underline needs to go the full length of the header, otherwise it renders incorrectly. Please fix the length.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dcbenito) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:305]: Please change this header to: radix_sort_by_key Out of Place Example 
The header underline needs to go the full length of the header, otherwise it renders incorrectly. Please fix the length.

**Review by dcbenito — CHANGES_REQUESTED** [review]
Minor fixes needed, otherwise LGTM.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:258]: The examples section became disproportionally large compared to the rest documentation. But still, I do not treat it too large to block the PR, so the suggestions below are optional or/and may be done separately.

There are several things to consider:

- Make the examples collapsible. https://github.com/executablebooks/sphinx-togglebutton may be useful.
- Move the example to the bottom of the page.
- Move the example to the separate files and provide links.

**Review by dmitriy-sobolev — COMMENTED** [review]
Thanks for the documentation update and fixes. I've left several comments.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:40]: Let's align the calls to follow that suggestion: https://github.com/oneapi-src/oneDPL/pull/1328#discussion_r1466631788.

```suggestion
   radix_sort (sycl::queue q, Range1&& r, Range2&& r_out,
               KernelParam param) // (3)
```

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:47]:
```suggestion
   radix_sort (sycl::queue q, Iterator1 first, Iterator1 last,
               Iterator2 first_out, KernelParam param) // (4)
```

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:71]:
```suggestion
   radix_sort_by_key (sycl::queue q, KeysRng1&& keys, ValsRng1&& values,
                      KeysRng2&& keys_out, ValsRng2&& vals_out,
                      KernelParam param) // (7)
```

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: fixed

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: fixed.  removed monospace in a number of headers and capitalized headers.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: fixed.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: fixed.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:259]: @dcbenito, is there any problem with using the collapsible examples?  I think I like that idea the best.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: done.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: done

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: done

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: No, "overload" looks fine.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:260]: I am also considering splitting that page in two parts: `radix_sort` and `radix_sort_by_key`. That will solve the issue. 

Let me try it. I will make a branch from this one: it should not take too much time. Then, we can decide if it worth continuing or we should do something different.

Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:260]: I'm experimenting with the toggle stuff, and setting up my environment to be able to do the build locally rather than relying on the artifacts from the runners.
We can compare the two options.

Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:260]: After struggling with getting sphinx togglebutton to render properly with a code block for a while, I don't think it is worth the trouble.  I also tried a custom css solution I found but that also has issues in combination with a code block.  

I'm sure there is a way to do it, but I think splitting the pages is the better option.  

Do you want to work to merge this PR first and then split? close this in favor of your combined PR, or target your PR at this branch?

Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:259]: Let's merge this PR first: this PR is valuable as it is, we will keep the changes isolated.

Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:259]: Meanwhile, I am working on splitting the page in #1586.

Comment (dcbenito) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst:259]: I was worried that togglebutton wouldn't work. I think splitting the page makes sense.

**Review by dcbenito — DISMISSED** [review]

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (adamfidel) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:104]: (Feedback based on #1559)
```suggestion
| ``q``                                         | The SYCL* queue where kernels are submitted.                        |
```

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst
Comment (adamfidel) [inline documentation/library_guide/kernel_templates/esimd/radix_sort.rst:107]: (Feedback based on https://github.com/oneapi-src/oneDPL/pull/1559)
```suggestion
|                                               | The sequences to apply the algorithm to.                            |
```

**Review by dmitriy-sobolev — APPROVED** [review]
LGTM

**Pull request merged**
